common/txmgr/types/tx.go

```diff
@@ -32,3 +37,75 @@ const (
 type Transaction interface {
 	GetID() string
 }
+
+type NewTx[ADDR types.Hashable, TX_HASH types.Hashable] struct {
```

Comment:
Curious as to why an address is a `types.Hashable`? It's not a hash of anything.

Reply:
We can actually use a better name.
Hashable is used for Address, TxHash, BlockHash etc. Basically anything that has unique value among others of same type.

I can think of some alternative names:
- Unique
- UniqueType
- Identifier
- UniqueIdentifier
- UniqueId

Reply:
I would agree that `Hashable` is an odd name - especially for `Address` types

we have a story to revisit the naming of generic types - I will mark these ideas in the ticket
https://smartcontract-it.atlassian.net/browse/BCI-1157